core/internal/leet/handlers.go

```diff
@@ -18,19 +18,22 @@ func (m *Model) processRecordMsg(msg tea.Msg) (*Model, tea.Cmd) {
 	}()
 
 	switch msg := msg.(type) {
+	case RunMsg:
+		m.logger.Debug("model: processing RunMsg")
+		m.leftSidebar.ProcessRunMsg(msg)
+		m.runState = RunStateRunning
+		if m.isLoading {
+			m.isLoading = false
+		}
```

Comment:
The `if` is redundant isn't it?


```suggestion
		m.isLoading = false
```